app/routes.py

```diff
@@ -162,7 +164,35 @@ def add_online_details():
     return render_template("create-masterclass/location/online-details.html")
 
 
-@main_bp.route("/create-masterclass/location/search", methods=["GET"])
+@main_bp.route("/create-masterclass/location/search", methods=["GET", "POST"])
 @login_required
 def search_for_location():
+    if request.method == "POST":
+        query = request.form["location"]
+        is_database_data = None
```

Comment:
I don't think you need this - there's no way `is_database_data` is anything but True or False by the end of this execution, because you've got if/else. You only really need to pre-set the variable if there's a chance it won't be set by the end, for example in an if/elif situation